docs/classic-ui/views.md

```diff
@@ -1,13 +1,1007 @@
+View methods are exposed to page templates and such, so you can also call
+view methods directly from a page template, not only from Python code.
+
+### More information
+
+- {doc}`Context helpers </develop/plone/misc/context>`
+- {doc}`Expressions </develop/plone/functionality/expressions>`
+
+### Historical perspective
+
+Often, the point of using helper views is that you can have reusable
+functionality which can be plugged in as one-line code around the system.
+Helper views also get around the following limitations:
+
+- TAL security.
+- Limiting Python expression to one line.
+- Not being able to import Python modules.
+
+```{Note}
+Using `RestrictedPython` scripts (creating Python through the
+Management Interface) and Zope 2 Extension modules is discouraged.
+The same functionality can be achieved with helper views, with less
+potential pitfalls.
+```
+
+## Reusing view template snippets or embedding another view
+
+To use the same template code several times you can either:
+
+- create a separate `BrowserView` for it and then call this view (see
+  [Accessing a view instance in code] below);
+- share a `ViewPageTemplate` instance between views and using it several
+  times.
+
+```{Note}
+The Plone 2.x way of doing this with TAL template language macros is
+discouraged as a way to provide reusable functionality in your add-on
+product.
+This is because macros are hardwired to the TAL template language, and
+referring to them outside templates is difficult.
+
+If you ever need to change the template language, or mix in other
+template languages, you can do better when templates are a
+feature of a pure Python based view, and not vice versa.
+```
+
+Here is an example of how to have a view snippet which can be used by
+subclasses of a base view class. Subclasses can refer to this template
+at any point of the view rendering, making it possible for subclasses
+to have fine-tuned control over how the template snippet is
+represented.
+
+Related Python code:
+
+```
+from Products.Five import BrowserView
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+
+class ProductCardView(BrowserView):
+    """
+    End user visible product card presentation.
+    """
+    implements(IProductCardView)
+
+    # Nested template which renders address box + buy button
+    summary_template = ViewPageTemplateFile("summarybox.pt")
+
+
+    def renderSummary(self):
+        """ Render summary box
+
+        @return: Resulting HTML code as Python string
+        """
+        return self.summary_template()
+```
+
+Then you can render the summary template in the main template associated
+with `ProductCardView` by calling the `renderSummary()` method and TAL
+non-escaping HTML embedding.
+
+```html
+<h1 tal:content="context/Title" />
+
+<div tal:replace="structure view/renderSummary" />
+
+<div class="description">
+    <div tal:condition="python:context.Description().decode('utf-8') != u'None'" tal:replace="structure context/Description" />
```

Comment:
Are we still using markers for Unicode strings in Python 3? Can this snippet be updated for Python 3?